<?xml version="1.0" encoding="utf-8"?>
<launch>

    <param name="/use_sim_time" type="bool" value="false"/>

    <rosparam command="load" file="$(find rnw_ros)/config/swarm_rnw_config.yaml"/>

    <node pkg="mocap_optitrack" type="mocap_node" name="mocap_node" respawn="false" launch-prefix="" required="true" output="log">
        <rosparam file="$(find mocap_optitrack)/config/mocap.yaml" command="load" />
    </node>

    <node pkg="mocap2odom" type="mocap2odom" name="mocap2odom" output="screen" >
        <param name="publish_uav_odom" type="bool" value="true"/>
        <rosparam file="$(find mocap2odom)/config/calib.yaml" command="load" />
    </node>

    <node pkg="rnw_ros" type="pub_cone_state_node" name="pub_cone_state_node" required="true" output="screen">
        <param name="cut_euler_velocity" type="bool" value="true"/>
        <param name="max_euler_velocity" value="10"/>
    </node>

    <node pkg="rnw_ros" type="visualize_cone_node" name="visualize_cone_node" output="screen" required="true">
        <param name="cone_color_r" type="double" value="0"/>
        <param name="cone_color_g" type="double" value="1"/>
        <param name="cone_color_b" type="double" value="0"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" output="log" required="false" args="-d $(find rnw_ros)/config/swarm.rviz"/>

    <node pkg="rnw_ros" type="pub_caging_state_node" name="pub_caging_state_node" output="screen" required="true"/>

    <node pkg="rqt_multiplot" type="rqt_multiplot" name="rqt_multiplot" output="log" required="true"
          args="--multiplot-config $(find rnw_ros)/config/rqt_multiplot_grip.xml --multiplot-run-all"/>

</launch>
